sys/test_utils/netdev_ieee802154_minimal/netdev_ieee802154_minimal.c

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (C) 2022 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for
+ * more details.
+ */
+
+/**
+ * @ingroup     test_utils_netdev_ieee802154_minimal
+ * @{
+ *
+ * @file
+ * @brief       Implementation of netdev IEEE 802.15.4 minimal test utility
+ *              module
+ *
+ * @author      Leandro Lanzieri <leandro.lanzieri@haw-hamburg.de>
+ */
+#include <stdio.h>
+
+#include "event.h"
+#include "event/thread.h"
+#include "od.h"
+#include "net/ieee802154.h"
+#include "net/l2util.h"
+#include "net/netdev.h"
+#include "net/netdev/ieee802154.h"
+#include "test_utils/netdev_ieee802154_minimal.h"
+#include "netdev_ieee802154_minimal_internal.h"
+
+/* provided by the test application */
+#include "init_dev.h"
+
+device_reg_entry_t _devices[NETDEV_IEEE802154_MINIMAL_NUMOF];
+static uint8_t _buffer[IEEE802154_FRAME_LEN_MAX];
+static char _addr_str[IEEE802154_LONG_ADDRESS_LEN * 3];
+
+void _recv(netdev_t *dev)
+{
+    uint8_t src[IEEE802154_LONG_ADDRESS_LEN], dst[IEEE802154_LONG_ADDRESS_LEN];
+    size_t mhr_len, data_len, src_len, dst_len;
+    netdev_ieee802154_rx_info_t rx_info;
+    le_uint16_t src_pan, dst_pan;
+
+    putchar('\n');
+    /* receive the frame */
+    data_len = dev->driver->recv(dev, _buffer, sizeof(_buffer), &rx_info);
```

Comment:
As pointed in #17813, we should have `ssize_t data_len` and check for errors while receiving (`data_len < 0`)

Reply:
+1. If the `recv` function fails, that will trigger a buffer overflow.